sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotHardcodeCredentials_DefaultValues.cs

```diff
@@ -63,6 +63,16 @@ public void StandardAPI(SecureString secureString, string nonHardcodedPassword,
             new PasswordDeriveBytes("hardcoded", byteArray, "strHashName", 1); // Noncompliant
             new PasswordDeriveBytes("hardcoded", byteArray, "strHashName", 1, cspParams); // Noncompliant
         }
+
+        public void CompliantParameterUse(string pwd)
+        {
+            string query1 = "password=?";
+            string query2 = "password=:password";
+            string query3 = "password=:param";
+            string query4 = "password='"+pwd+"'";
+            string query5 = "password={0}";
+            string query6 = @"Server=myServerName\myInstanceName;Database=myDataBase;Password=:myPassword;User Id=:username;";
+        }
```

Comment:
I suggest to add (crazy) test like this to document FN and explain the semicolon handling and :param being valid
`"password=':crazy;secret';user=xxx"`